packages/protocol/contracts/nft_bridge/Erc721Bridge.sol

```diff
@@ -0,0 +1,283 @@
+// SPDX-License-Identifier: MIT
+//  _____     _ _         _         _
+// |_   _|_ _(_) |_____  | |   __ _| |__ ___
+//   | |/ _` | | / / _ \ | |__/ _` | '_ (_-<
+//   |_|\__,_|_|_\_\___/ |____\__,_|_.__/__/
+
+pragma solidity ^0.8.20;
+
+import { AddressResolver } from "../common/AddressResolver.sol";
+import { EssentialContract } from "../common/EssentialContract.sol";
+import { Proxied } from "../common/Proxied.sol";
+import { IErc721Bridge } from "./erc721/IErc721Bridge.sol";
+import { NftBridgeErrors } from "./NftBridgeErrors.sol";
+import { LibErc721BridgeData } from "./erc721/libs/LibErc721BridgeData.sol";
+import { LibErc721BridgeProcess } from "./erc721/libs/LibErc721BridgeProcess.sol";
+import { LibErc721BridgeRelease } from "./erc721/libs/LibErc721BridgeRelease.sol";
+import { LibErc721BridgeSend } from "./erc721/libs/LibErc721BridgeSend.sol";
+import { LibErc721BridgeStatus } from "./erc721/libs/LibErc721BridgeStatus.sol";
+
+/**
+ * This contract is an ERC-721 token bridge contract which is deployed on both L1 and L2.
+ * which calls the library implementations. See _IErc721Bridge_ for more details.
+ * @dev The code hash for the same address on L1 and L2 may be different.
+ * @custom:security-contact hello@taiko.xyz
+ */
+contract Erc721Bridge is EssentialContract, IErc721Bridge, NftBridgeErrors {
```

Comment:
There should just be one Bridge, Bridge is a common infrastructure for messaging, it's agnostic to message types.